### PR TITLE
feat: support layers 

### LINKED ISSUE #18: Draw "active" nodes/links on top of other nodes/links
image

In the screenshot above, the connection between the "small3" and "med1" nodes appears behind some other connection.

It might be nice to have the currently active nodes/links drawn on "top of" everything else.

To accomplish this, we'd have to essentially emulate a "z-index", sort our drawables by that "z-index" property and then render things with a higher "z-index" last.

### PR DESCRIPTION
**Pull request by jollyjerr**
This pr closes #18

https://github.com/jollyjerr/mindgraph/assets/51095001/b77732ec-0f26-458b-9af4-01534642cd72

### CODE REVIEW

File: packages/draw/src/canvas.ts [hunk 1]
```diff
@@ -100,36 +108,57 @@ export class Canvas {
...
-    drawables.forEach((d) => {
-      const highlight = activeDrawables.includes(d)
-        ? 'active'
-        : activeDrawables.length > 0
-        ? 'dimmed'
-        : 'normal';
-      d.draw(this, highlight);
-    });
+    if (config.highlight === 'dimmed') {
+      this.context.globalAlpha = 0.4;
```
Comment (mpryor) [inline packages/draw/src/canvas.ts:139]: Should this be externalized somewhere? Maybe via the styles config?

Comment (jollyjerr) [reply packages/draw/src/canvas.ts:142]: Yes for sure!

File: packages/draw/src/artist.ts [hunk 2]
```diff
@@ -18,23 +24,27 @@ export class Artist {
...
+      initialWidth: this.visual_canvas.node()?.width || 0,
+      initialHeight: this.visual_canvas.node()?.height || 0,
+    };
 
-  public getStyles(): Styles {
-    return this.styles;
+    this.drawables = [];
+    this.baseLayer = {
+      drawables: [],
+      canvas: new Canvas(undefined, memoryCanvasConfig),
```
[the remark this answers is not in the document]
Comment (jollyjerr) [reply packages/draw/src/artist.ts]: I think that is the way to go for now. I was originally picturing an API where you pass in an arbitrary number of canvases that are each like a Photoshop layer and then visual_canvas just draws the layers, but I think if we can replicate the functionality of layers on a single canvas that will be better

**Review by mpryor — APPROVED** [review]
Looks good! Thanks for separating out the renderables to separate files!

**Pull request merged**